### ISSUE TITLE
Run `make dep` strictly before `make $(TARGET)` puzzle

### ISSUE THREAD

**Issue by cschol** [VCVRack/Fundamental#40]
Compilation from a **fresh checkout** fails when using parallelized build with the `-j` argument.

Maybe the problem is, that the dependencies are build *in parallel* with the code files and some code files may need the dependencies, but they are not build yet?

Running just `make` works fine, running `make -j 4` does not.

@ASCustomWorks has the same problem as he is using the same `Makefile` recipe to build his dependencies.

**Comment by AScustomWorks** [comment]
Indeed, I updated the 2 delay modules to match Fundamental's code and makefile too.
Just tried make -j4, got this error:

```src/DelayPlus.cpp:10:10: fatal error: 'samplerate.h' file not found #include "samplerate.h"```

Is just matter of building the dependencies first, right? (but makefile changes are way over my skills) This way is neater than make dep, make though.

**Comment by AndrewBelt** [comment]
Try 03c14d5bb6b6076a91070b0b90c8366cdd50abbc in Rack and c9142a85b0edde50ed300a76552ff4dc0a4cb83b in Fundamental
This makes `dep` a target of all object files.

**Comment by cschol** [comment]
Sorry, it doesn't work. I didn't do a pull on `Fundamental` before I tested.

**Comment by cschol** [comment]
Looks like a botched merge? For some reason 343b66620230199177d4235b26af056fabb6baa9 did not make it in the final `Makefile` version.

```
cd dep && curl -OL http://www.mega-nerd.com/SRC/libsamplerate-0.1.9.tar.gz
/bin/sh: 1: cd: can't cd to dep
Makefile:18: recipe for target 'dep/lib/libsamplerate.a' failed
make: *** [dep/lib/libsamplerate.a] Error 2
```

**Comment by cschol** [comment]
And `Fundamental` plugin doesn't load:

```
[0.032 warn] Failed to load library ./plugins/Fundamental/plugin.so: ./plugins/Fundamental/plugin.so: undefined symbol: src_delete
```

**Comment by AndrewBelt** [comment]
Try 1878e96

**Comment by cschol** [comment]
Builds **and** loads for both plugins now. Thank you!

**Comment by cschol** [comment]
@AndrewBelt The `Makefile` is still a problem. It does not create the `dep` directory if it is not already there. Fails on all platforms if `dep` is not there. Take a look at the `Makefile` in the latest revision. The merge took out the `$(shell mkdir -p dep)` line although on of the merge commits had it in there.

**Comment by AndrewBelt** [comment]
Fixed in 0543c35

`make clean` should not remove dependencies.

**Comment by cschol** [comment]
Agreed on `make clean`. I ran the wrong command and it did not remove `dep`.

Sorry, this does not work. The variable `CONFIGURE` is initialized when `dep` does not exist yet, so the ``--prefix` is wrong, because `realpath` on a non-existing directory returns an empty string. 

You could remove the ":" from `CONFIGURE` in `dep.mk` to defer initialization until it is used in Fundamental's `Makefile`.

**Comment by AndrewBelt** [comment]
aaf7473

**Comment by cschol** [comment]
@AndrewBelt Oh, yes. That's a better fix. Tested on Linux and it works.

Now, on Windows, `libsamplerate` compilation fails. This seems to be a different issue (possibly related to `libsamplerate` itself) and I'll open a separate issue to track.

**Comment by cschol** [comment]
The Windows compilation issue is same as #38.
